packages/vite/src/node/build.ts

```diff
@@ -297,14 +297,15 @@ export function resolveBuildPlugins(config: ResolvedConfig): {
   post: Plugin[]
 } {
   const options = config.build
-
+  const { commonjsOptions } = options
+  const usePluginCommonjs =
+    !Array.isArray(commonjsOptions?.include) ||
+    commonjsOptions?.include.length !== 0
   return {
     pre: [
       ...(options.watch ? [ensureWatchPlugin()] : []),
       watchPackageDataPlugin(config),
-      ...(config.legacy?.buildRollupPluginCommonjs
-        ? [commonjsPlugin(options.commonjsOptions)]
-        : []),
+      ...(usePluginCommonjs ? [commonjsPlugin(options.commonjsOptions)] : []),
```

Comment:
Shouldn't we enable the commonjs plugin when `optimizeDeps.disabled === 'build'` (or `true`) instead?

Reply:
I'm thinking that users may want to use optimization for some deps, and plugin-commonjs for some others. 
You could say
```
build: {
	commonjsOptions: { include: ['dep-with-issue' ] 
}, 
optimizeDeps: { exclude: ['dep-with-issue'] }`
```

Reply:
Ah ok I think that works too. The condition tripped me up 😅